db/migrate/20171213161035_add_indexes_on_images.rb

```diff
@@ -0,0 +1,23 @@
+class AddIndexesOnImages < ActiveRecord::Migration[5.0]
+  def up
+    deduped = Image.group(:name, :compute_resource_id, :operatingsystem_id).pluck('min(id)')
+    Image.where.not(id: deduped).each_with_index do |image, i|
+      say "Found images with duplicate name #{image.name} on Compute Resource #{image.compute_resource}!"
+      say "Duplicates have been renamed, please check your setup to resolve the duplication."
+      image.update_column(:name, "#{image.name}(Duplicate #{i})")
+    end
+    deduped = Image.group(:uuid, :compute_resource_id).pluck('min(id)')
+    Image.where.not(id: deduped).each_with_index do |image, i|
+      say "Found images with duplicate uuid #{image.uuid} on Compute Resource #{image.compute_resource}!"
+      say "Duplicates' uuids have been modified, please check your setup to resolve the duplication."
+      image.update_column(:uuid, "#{image.uuid}(Duplicate #{i})")
```

Comment:
Is the column wide enough for this? Also wondering if on postgres the native uuid column type is used which would probably not like this.

Reply:
yes, it's a string column currently with limit of 255 so should be no problem.